kernel/kernel-api/src/main/java/io/delta/kernel/internal/files/ParsedLogData.java

```diff
@@ -93,16 +88,18 @@ public static ParsedLogData forFileStatus(FileStatus fileStatus) {
 
   public static ParsedLogData forInlineData(
       long version, ParsedLogType type, ColumnarBatch inlineData) {
-    if (type == ParsedLogType.PUBLISHED_DELTA || type == ParsedLogType.RATIFIED_STAGED_COMMIT) {
-      throw new IllegalArgumentException(
-          "For PUBLISHED_DELTA|RATIFIED_STAGED_COMMIT, use ParsedLogData.forFileStatus() instead");
-    } else if (type == ParsedLogType.LOG_COMPACTION) {
-      throw new IllegalArgumentException(
-          "For LOG_COMPACTION, use ParsedLogCompactionData.forInlineData() instead");
-    } else if (type.category == ParsedLogCategory.CHECKPOINT) {
-      return ParsedCheckpointData.forInlineData(version, type, inlineData);
+    switch (type.category) {
+      case LOG_COMPACTION:
+        throw new IllegalArgumentException(
```

Comment:
Mmm. Sorry I missed this before. I feel like we should rather design the API so this can't happen. My initial thought is, we have `ParsedCheckpointData`, so could we just remove the `type` argument, and then you just have to call on the type you want directly? Perhaps method names change to indicate what type you're getting too.

Reply:
I can go ahead and create that, SGTM!

Reply:
update: we also don't have a `ParsedDeltaData` class today (though, I think I'd like to add one so it can contain the additional commitTimestamp info that is exclusive to delta files and commits)

for now, I'll just add these as static factories on ParsedLogData for the delta and checksum types

Reply:
I decided just to rip off the bandaid and do a larger refactor; now there is one `ParsedLogData.forFileStatus` method that will delegate to children, and there is no `ParsedLogData.forInlineData` method, and instead each child class implements that method themselves.